File: app/src/main/java/com/mstc/mstcapp/ui/resources/ViewPagerAdapter.java
package com.mstc.mstcapp.ui.resources;


import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentPagerAdapter;

import com.mstc.mstcapp.ui.resources.detailsTab.DetailsFragment;
import com.mstc.mstcapp.ui.resources.resourceTab.ResourceTabFragment;
import com.mstc.mstcapp.ui.resources.roadmapTab.RoadmapTabFragment;

public class ViewPagerAdapter extends FragmentPagerAdapter {

    private final String domain;

    public ViewPagerAdapter(@NonNull FragmentManager fm, String domain) {
        super(fm);
        this.domain = domain;
    }

    @NonNull
    @Override
    public Fragment getItem(int position) {
        if (position == 0)
            return new DetailsFragment(domain);
        else if (position == 1)
            return new RoadmapTabFragment(domain);
        else
            return new ResourceTabFragment(domain);
    }

    @Override
    public int getCount() {
        return 3;
    }

    @Override
    public CharSequence getPageTitle(int position) {
        if (position == 0)
            return "Details";
        else if (position == 1)
            return "Roadmap";
        else
            return "Resources";
    }

}

File: app/src/main/java/com/mstc/mstcapp/ui/resources/ViewPagerAdapter.kt
package com.mstc.mstcapp.ui.resources

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter
import com.mstc.mstcapp.ui.resources.details.DetailsFragment
import com.mstc.mstcapp.ui.resources.resource.ResourceFragment
import com.mstc.mstcapp.ui.resources.roadmap.RoadmapFragment

class ViewPagerAdapter(fm: FragmentManager, private val domain: String) :
    FragmentPagerAdapter(fm) {
    override fun getItem(position: Int): Fragment {
        return when (position) {
            0 -> DetailsFragment(domain)
            1 -> RoadmapFragment(domain)
            else -> ResourceFragment(domain)
        }
    }

    override fun getCount(): Int {
        return 3
    }

    override fun getPageTitle(position: Int): CharSequence? {
        return when (position) {
            0 -> "Details"
            1 -> "Roadmap"
            else -> "Resources"
        }
    }
}
